test/testdata/parser/quoted_symbol_bug.rb

```diff
@@ -0,0 +1,8 @@
+# typed: true
+
+class A < T::Struct
+  prop :'foo', Integer
+end
+
+a = A.new(foo: 0)
+a.foo = '' # error: Assigning a value to `'foo'` that does not match expected type `Integer`
```

Comment:
The error message still mentions the `'...'`, even though the user is assigning to `foo`, not `'foo'`.

Do you want to try to make it show ``value to `foo` that does not match`` in this error message?

Reply:
ack, I'll TAL